### PR TITLE
Fix cache level size info for systems data

### PR DESCRIPTION
**Pull request by achawla2012**
### Proposed changes

- To get cpuInfo,  execute  lscpu command on host and read its output.  (as nginx-amplify-agent currently [does](https://github.com/nginxinc/nginx-amplify-agent/blob/fa4fb352ee6db134899364f5c8a3e0f1bd536ed3/amplify/agent/collectors/system/meta.py#L92))
- If read fails, then fall back to current nginx-agent implementation.
- In addition to it, cache will have formatted bytes string instead of just bytes.  (as nginx-amplify-agent currently [does](https://github.com/nginxinc/nginx-amplify-agent/blob/fa4fb352ee6db134899364f5c8a3e0f1bd536ed3/amplify/agent/collectors/system/meta.py#L109))
   **Formatted bytes.** -> E.g. instead of "L1" : "1024", cache map will be represented by  "L1" : "1 KiB"

#### Test  -
- Added Units tests.
- Deployed agent changes in sandbox. Validated change by intercepting message that converter sends to receiver (aka AMP backend).

#### Callout -
Testing changes on different archs.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [X] If applicable, I have added tests that prove my fix is effective or that my feature works
- [X] If applicable, I have checked that any relevant tests pass after adding my changes
- [X] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

### CODE REVIEW

**Comment by oliveromahony** [comment]
please run ```make deps``` but this looks good

**Comment by achawla2012** [comment]
> please run `make deps` but this looks good

`make deps` results in failure.

```
Warning: protobuf 21.12 is already installed and up-to-date.
To reinstall 21.12, run:
  brew reinstall protobuf
```

```
-> echo $PATH                                                              
/usr/local/go:/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:**/usr/local/go/bin**:/Applications/Visual Studio Code.app/Contents/Resources/app/bin:/Applications/Visual Studio Code.app/Contents/Resources/app/bin

```

```
make install-tools

fatal: No names found, cannot describe anything.
Getting Tools
go get github.com/alvaroloes/enumer
go get github.com/gogo/protobuf/protoc-gen-gogo
go get github.com/gogo/protobuf/protoc-gen-gogofast
go get github.com/golang/mock/mockgen
go get github.com/maxbrunsfeld/counterfeiter/v6
go get github.com/mwitkow/go-proto-validators/protoc-gen-govalidators
go get google.golang.org/grpc/cmd/protoc-gen-go-grpc
go get github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc
go: module github.com/golang/protobuf is deprecated: Use the "google.golang.org/protobuf" module instead.
go get github.com/go-swagger/go-swagger/cmd/swagger
go get github.com/bufbuild/buf/cmd/buf
go get github.com/goreleaser/nfpm/v2/cmd/nfpm
Installing Tools
go install github.com/alvaroloes/enumer
go install github.com/gogo/protobuf/protoc-gen-gogo
go install github.com/gogo/protobuf/protoc-gen-gogofast
go install github.com/golang/mock/mockgen
go install github.com/maxbrunsfeld/counterfeiter/v6
go install github.com/mwitkow/go-proto-validators/protoc-gen-govalidators
go install google.golang.org/grpc/cmd/protoc-gen-go-grpc
go install github.com/pseudomuto/protoc-gen-doc/cmd/protoc-gen-doc
go install github.com/go-swagger/go-swagger/cmd/swagger
go install github.com/bufbuild/buf/cmd/buf
go install github.com/goreleaser/nfpm/v2/cmd/nfpm
```

```
make deps

fatal: No names found, cannot describe anything.
cd sdk && go mod tidy && go mod vendor &&  make generate && go mod tidy && go mod vendor
go: finding module for package github.com/nginx/agent/sdk/v2/proto/events
go: finding module for package github.com/nginx/agent/sdk/v2/proto
github.com/nginx/agent/sdk/v2 imports
	github.com/nginx/agent/sdk/v2/proto: no matching versions for query "latest"
github.com/nginx/agent/sdk/v2/client imports
	github.com/nginx/agent/sdk/v2/proto/events: no matching versions for query "latest"
make: *** [deps] Error 1
```

**Comment by oliveromahony** [comment]
Can you provide the output for 4 operating systems from our support matrix? https://github.com/nginx/agent/blob/main/Makefile#L12
Ubuntu 22, RHEL 8, SUSE 15 and FreeBSD 13 thanks

**Comment by oliveromahony** [comment]
Can you provide the output for 4 operating systems from our support matrix? https://github.com/nginx/agent/blob/main/Makefile#L12
Ubuntu 22, RHEL 8, SUSE 15 and FreeBSD 13 thanks

**Comment by achawla2012** [comment]
> Can you provide the output for 4 operating systems from our support matrix? https://github.com/nginx/agent/blob/main/Makefile#L12 Ubuntu 22, RHEL 8, SUSE 15 and FreeBSD 13 thanks

Find O/P for SUSE, RHEL, Ubuntu.
`agent.log` corresponds to `env.host` field. It reflects corresponding fields populating with a fix.

### Ubuntu 22
```
ubuntu@ip-172-16-0-49:~$ lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         48 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  2
  On-line CPU(s) list:   0,1
Vendor ID:               AuthenticAMD
  Model name:            AMD EPYC 7571
    CPU family:          23
    Model:               1
    Thread(s) per core:  2
    Core(s) per socket:  1
    Socket(s):           1
    Stepping:            2
    BogoMIPS:            4400.00
    Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 
                         sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch topoext vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 clzero xsaveer
                         ptr arat npt nrip_save
Virtualization features: 
  Hypervisor vendor:     KVM
  Virtualization type:   full
Caches (sum of all):     
  L1d:                   32 KiB (1 instance)
  L1i:                   64 KiB (1 instance)
  L2:                    512 KiB (1 instance)
  L3:                    8 MiB (1 instance)
NUMA:                    
  NUMA node(s):          1
  NUMA node0 CPU(s):     0,1
Vulnerabilities:         
  Itlb multihit:         Not affected
  L1tf:                  Not affected
  Mds:                   Not affected
  Meltdown:              Not affected
  Mmio stale data:       Not affected
  Retbleed:              Mitigation; untrained return thunk; SMT vulnerable
  Spec store bypass:     Vulnerable
  Spectre v1:            Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:            Mitigation; Retpolines, STIBP disabled, RSB filling, PBRSB-eIBRS Not affected
  Srbds:                 Not affected
  Tsx async abort:       Not affected
```

```
ubuntu@ip-172-16-0-49:~$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

```
//agent .log
"{\"agent\":\"\",\"boot\":1677788139,\"hostname\":\"ip-172-16-0-49\",\"display_name\":\"ip-172-16-0-49\",\"os-type\":\"linux\",\"uuid\":\"1325a622-f367-3f34-b907-a85d183844e8\",\"uname\":\"Linux ip-172-16-0-49 5.15.0-1030-aws #34-Ubuntu SMP Mon Jan 23 20:13:32 UTC 2023 x86_64\",\"disk_partitions\":[{\"mountpoint\":\"/\",\"device\":\"/dev/nvme0n1p1\",\"fstype\":\"ext4\"},{\"mountpoint\":\"/snap/amazon-ssm-agent/6312\",\"device\":\"/dev/loop0\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/core20/1822\",\"device\":\"/dev/loop2\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/snapd/18357\",\"device\":\"/dev/loop4\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/lxd/24322\",\"device\":\"/dev/loop3\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/snap/core18/2679\",\"device\":\"/dev/loop1\",\"fstype\":\"squashfs\"},{\"mountpoint\":\"/boot/efi\",\"device\":\"/dev/nvme0n1p15\",\"fstype\":\"vfat\"},{\"mountpoint\":\"/snap/core18/2697\",\"device\":\"/dev/loop5\",\"fstype\":\"squashfs\"}],\"network\":{\"interfaces\":[{\"mac\":\"\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"::1\"}],\"ipv4\":[{\"prefixlen\":8,\"netmask\":\"255.0.0.0\",\"address\":\"127.0.0.1\"}],\"name\":\"lo\"},{\"mac\":\"0a:2b:71:5b:33:33\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"2600:1f13:a98:5701:a8ea:8a0e:327:a8b7\"},{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"fe80::82b:71ff:fe5b:3333\"}],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"172.16.0.49\"}],\"name\":\"ens5\"},{\"mac\":\"02:42:af:f1:90:b4\",\"ipv6\":[],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"192.0.10.1\"}],\"name\":\"docker0\"}],\"default\":\"lo\"},\"processor\":[{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2200,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32 KiB (1 instance)\",\"L1i\":\"64 KiB (1 instance)\",\"L2\":\"512 KiB (1 instance)\",\"L3\":\"8 MiB (1 instance)\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}},{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2200,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32 KiB (1 instance)\",\"L1i\":\"64 KiB (1 instance)\",\"L2\":\"512 KiB (1 instance)\",\"L3\":\"8 MiB (1 instance)\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}}],\"release\":{\"codename\":\"jammy\",\"id\":\"ubuntu\",\"name\":\"Ubuntu\",\"version_id\":\"22.04\",\"version\":\"22.04.2 LTS (Jammy Jellyfish)\"},\"tags\":[],\"agent_accessible_dirs\":\"/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules:/etc/nms\"}"
```

### RHEL
```
[ec2-user@ip-172-16-0-214 ~]$ lscpu
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  2
Core(s) per socket:  1
Socket(s):           1
NUMA node(s):        1
Vendor ID:           AuthenticAMD
CPU family:          23
Model:               1
Model name:          AMD EPYC 7571
Stepping:            2
CPU MHz:             2199.522
BogoMIPS:            4399.04
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           64K
L2 cache:            512K
L3 cache:            8192K
NUMA node0 CPU(s):   0,1
```

```
[ec2-user@ip-172-16-0-214 ~]$ cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="8.7 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.7"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.7 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.7
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.7" 
```

```
// Agent log.
"{\"agent\":\"\",\"boot\":1677651430,\"hostname\":\"ip-172-16-0-214.us-west-2.compute.internal\",\"display_name\":\"ip-172-16-0-214.us-west-2.compute.internal\",\"os-type\":\"linux\",\"uuid\":\"42721515-dc58-381e-b208-b806af379e51\",\"uname\":\"Linux ip-172-16-0-214.us-west-2.compute.internal 4.18.0-425.10.1.el8_7.x86_64 #1 SMP Wed Dec 14 16:00:01 EST 2022 x86_64\",\"disk_partitions\":[{\"mountpoint\":\"/\",\"device\":\"/dev/nvme0n1p2\",\"fstype\":\"xfs\"}],\"network\":{\"interfaces\":[{\"mac\":\"\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"::1\"}],\"ipv4\":[{\"prefixlen\":8,\"netmask\":\"255.0.0.0\",\"address\":\"127.0.0.1\"}],\"name\":\"lo\"},{\"mac\":\"02:3d:1b:71:ac:af\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"2600:1f13:553:8001:9426:65e7:ad32:c6b0\"},{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"fe80::3d:1bff:fe71:acaf\"}],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"172.16.0.214\"}],\"name\":\"eth0\"}],\"default\":\"lo\"},\"processor\":[{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.522,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}},{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.522,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}}],\"release\":{\"codename\":\"linux\",\"id\":\"rhel\",\"name\":\"Red Hat Enterprise Linux\",\"version_id\":\"8.7\",\"version\":\"8.7 (Ootpa)\"},\"tags\":[],\"agent_accessible_dirs\":\"/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules:/etc/nms\"}" 
```

### SUSE

```
ec2-user@ip-172-16-0-207:~> lscpu
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
Address sizes:       48 bits physical, 48 bits virtual
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  2
Core(s) per socket:  1
Socket(s):           1
NUMA node(s):        1
Vendor ID:           AuthenticAMD
CPU family:          23
Model:               1
Model name:          AMD EPYC 7571
Stepping:            2
CPU MHz:             2199.790
BogoMIPS:            4399.58
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           64K
L2 cache:            512K
L3 cache:            8192K
NUMA node0 CPU(s):   0,1
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch topoext vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 clzero xsaveerptr arat npt nrip_save
```
```
ec2-user@ip-172-16-0-207:~> cat /etc/os-release 
NAME="SLES"
VERSION="15-SP2"
VERSION_ID="15.2"
PRETTY_NAME="SUSE Linux Enterprise Server 15 SP2"
ID="sles"
ID_LIKE="suse"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:suse:sles:15:sp2"
```

```
//agent log
"{\"agent\":\"\",\"boot\":1677783989,\"hostname\":\"ip-172-16-0-207\",\"display_name\":\"ip-172-16-0-207\",\"os-type\":\"linux\",\"uuid\":\"d1af3bda-e1dd-351d-85e7-0784aa7f7fe0\",\"uname\":\"Linux ip-172-16-0-207 5.3.18-24.96-default #1 SMP Mon Nov 22 08:33:36 UTC 2021 (9a413cc) x86_64\",\"disk_partitions\":[{\"mountpoint\":\"/\",\"device\":\"/dev/nvme0n1p3\",\"fstype\":\"xfs\"},{\"mountpoint\":\"/boot/efi\",\"device\":\"/dev/nvme0n1p2\",\"fstype\":\"vfat\"}],\"network\":{\"interfaces\":[{\"mac\":\"\",\"ipv6\":[{\"prefixlen\":128,\"netmask\":\"ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff\",\"address\":\"::1\"}],\"ipv4\":[{\"prefixlen\":8,\"netmask\":\"255.0.0.0\",\"address\":\"127.0.0.1\"}],\"name\":\"lo\"},{\"mac\":\"02:21:3a:31:34:d5\",\"ipv6\":[{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"2600:1f13:843:6101:9691:3ac9:6f9:5a2d\"},{\"prefixlen\":64,\"netmask\":\"ffff:ffff:ffff:ffff::\",\"address\":\"fe80::21:3aff:fe31:34d5\"}],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"172.16.0.207\"}],\"name\":\"eth0\"},{\"mac\":\"02:42:56:7e:ca:ca\",\"ipv6\":[],\"ipv4\":[{\"prefixlen\":24,\"netmask\":\"255.255.255.0\",\"address\":\"192.0.10.1\"}],\"name\":\"docker0\"}],\"default\":\"lo\"},\"processor\":[{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.79,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}},{\"model\":\"1\",\"cores\":1,\"architecture\":\"x86_64\",\"mhz\":2199.79,\"hypervisor\":\"\",\"cpus\":2,\"virtualization\":\"\",\"cache\":{\"Cacheline bytes:\":\"64\",\"Features:\":\"ADX,AESNI,AVX,AVX2,AVXSLOW,BMI1,BMI2,CLMUL,CLZERO,CMOV,CMPXCHG8,CX16,F16C,FMA3,FXSR,FXSROPT,HTT,HYPERVISOR,LAHF,LZCNT,MMX,MMXEXT,MOVBE,NX,OSXSAVE,POPCNT,RDRAND,RDSEED,RDTSCP,SCE,SHA,SSE,SSE2,SSE3,SSE4,SSE42,SSE4A,SSSE3,X87,XGETBV1,XSAVE,XSAVEC,XSAVEOPT\",\"L1d\":\"32K\",\"L1i\":\"64K\",\"L2\":\"512K\",\"L3\":\"8192K\",\"SIMD 2:\":\"Streaming SIMD 2 Extensions\"}}],\"release\":{\"codename\":\"linux\",\"id\":\"sles\",\"name\":\"SLES\",\"version_id\":\"15.2\",\"version\":\"15-SP2\"},\"tags\":[],\"agent_accessible_dirs\":\"/etc/nginx:/usr/local/etc/nginx:/usr/share/nginx/modules:/etc/nms\"}"
```

**Comment by achawla2012** [comment]
> Can you provide the output for 4 operating systems from our support matrix? https://github.com/nginx/agent/blob/main/Makefile#L12 Ubuntu 22, RHEL 8, SUSE 15 and FreeBSD 13 thanks

### FreeBSD
As I spinned freebsd instance on testenv with datapath_os = `freebsd`, I am seeing lot of error in `agent.log` ( without my changes..) 
I would still see the same error even if I create another new instance. 
```
time="2023-03-03T20:46:08Z" level=error msg="Error reading exe information for process: 2064 error: not implemented yet"
time="2023-03-03T20:46:08Z" level=error msg="Error reading exe information for process: 2064 error: not implemented yet"
time="2023-03-03T20:46:08Z" level=error msg="Error reading exe information for process: 2064 error: not implemented yet"
...
time="2023-03-04T00:03:42Z" level=error msg="Error reading exe information for process: 2064 error: not implemented yet"
time="2023-03-04T00:03:42Z" level=error msg="Error reading exe information for process: 2064 error: not implemented yet"
...
time="2023-03-04T00:03:43Z" level=error msg="Error reading exe information for process: 2064 error: not implemented yet"
time="2023-03-04T00:03:43Z" level=error msg="Error reading exe information for process: 2064 error: not implemented yet"
```

```
ec2-user@ip-172-16-0-171:~ $ ps aux | grep 2064
root     2064   0.0  0.2  19588  8816  -  Is   20:36     0:00.00 nginx: master process /usr/local/sbin/nginx
```

**Comment by oliveromahony** [comment]
We have a fix for some of the issues you highlighted here: https://github.com/nginx/agent/pull/232
 
I added some comments to this PR

File: src/core/environment.go [hunk 1]
```diff
@@ -510,22 +510,82 @@ func processors() (res []*proto.CpuInfo) {
...
+	}
+	return cache
+}
+
+func getProcessorCacheInfo(cpuInfo cpuid.CPUInfo) map[string]string {
+	cache := getDefaultProcessorCacheInfo(cpuInfo)
+
+	out, err := exec.Command("lscpu").Output()
+	if err != nil {
+		log.Warnf("Error executing lscpu on host: %v", err)
```
Comment (oliveromahony) [inline src/core/environment.go:525]: can you update the warning to say please install lscpu in order to collect processor cache information - this is not installed on FreeBSD by default

Comment (oliveromahony) [reply src/core/environment.go:525]: If you can test the lscpu output on FreeBSD and the code works with it, I think this PR is good then

**Comment by achawla2012** [comment]
@oliveromahony messed up with rebase locally. I will close this PR.  

I cloned it to new MR here :- https://github.com/nginx/agent/pull/238

File: src/core/environment.go (hunk 1 above)
Comment (achawla2012) [reply src/core/environment.go:525]: @oliveromahony At the moment there is a bug where Nginx agent could not be installed on FreeBsd. 
Please suggest on how should I proceed with testing?

**Pull request closed without merging**